addons/web/static/tests/owl_compatibility_tests.js

```diff
@@ -0,0 +1,535 @@
+                }
+            }
+            Parent.template = xml`
+                <div>
+                    <ComponentAdapter Component="MyWidget" a1="'Hello'" a2="'World'"/>
+                </div>`;
+            Parent.components = { ComponentAdapter };
+
+            const target = testUtils.prepareTarget();
+            const parent = new Parent();
+            try {
+                await parent.mount(target);
+            } catch (e) {
+                assert.strictEqual(e.toString(),
+                    `Error: ComponentAdapter has more than 1 argument, 'widgetArgs' must be overriden.`);
+            }
+
+            parent.destroy();
+        });
+
+        QUnit.test("sub widget with several arguments (specific Adapter)", async function (assert) {
+            assert.expect(1);
+
+            const MyWidget = Widget.extend({
+                init: function (parent, a1, a2) {
+                    this._super.apply(this, arguments);
+                    this.a1 = a1;
+                    this.a2 = a2;
+                },
+                start: function () {
+                    this.$el.text(`${this.a1} ${this.a2}!`);
+                }
+            });
+            class MyWidgetAdapter extends ComponentAdapter {
+                get widgetArgs() {
+                    return [this.props.a1, this.props.a2];
+                }
+            }
+            class Parent extends Component {
+                constructor() {
+                    super(...arguments);
+                    this.MyWidget = MyWidget;
+                }
+            }
+            Parent.template = xml`
+                <div>
+                    <MyWidgetAdapter Component="MyWidget" a1="'Hello'" a2="'World'"/>
+                </div>`;
+            Parent.components = { MyWidgetAdapter };
+
+            const target = testUtils.prepareTarget();
+            const parent = new Parent();
+            await parent.mount(target);
+
+            assert.strictEqual(parent.el.innerHTML, '<div>Hello World!</div>');
+
+            parent.destroy();
+        });
+
+        QUnit.test("sub widget is updated when props change", async function (assert) {
+            assert.expect(2);
+
+            const MyWidget = Widget.extend({
+                init: function (parent, name) {
+                    this._super.apply(this, arguments);
+                    this.name = name;
+                },
+                start: function () {
+                    this.render();
+                },
+                render: function () {
+                    this.$el.text(`Hello ${this.name}!`);
+                },
+                update: function (name) {
+                    this.name = name;
+                },
+            });
+            class MyWidgetAdapter extends ComponentAdapter {
+                updateWidget(nextProps) {
+                    this.widget.update(nextProps.name);
+                    return this.widget.render.bind(this.widget);
```

Comment:
it would make more sense (to me) if this was in the `update` method of `MyWidget`